app/src/main/java/com/example/mobile_development/MainActivity.kt

```diff
@@ -0,0 +1,244 @@
+            Image(
+                modifier = Modifier
+                    .size(width = 500.dp, height = 800.dp)
+                    .padding(0.dp),
+                imageVector = ImageVector.vectorResource(id = R.drawable.ethereal_artefacts_logo),
+                contentDescription = "background_image",
+                contentScale = ContentScale.Inside
+            )
+        }
+
+        Row(
+            modifier = Modifier
+                .size(width = 400.dp, height = 70.dp)
+                .padding(10.dp)
+        ) {
+            Text(
+                text = "Log in",
+                modifier = modifier,
+                fontSize = 30.sp,
+                fontWeight = FontWeight.Bold,
+            )
+        }
+
+        Row(
+            modifier = Modifier
+                .size(width = 400.dp, 100.dp)
+                .padding(10.dp)
+        ) {
+            OutlinedTextField(
+                modifier = modifier
+                    .size(width = 400.dp, height = 50.dp)
+                    .fillMaxSize(),
+                value = email,
+                onValueChange = { newValue -> setEmail(newValue) },
+                placeholder = { Text("Email") },
+                isError = !ValidateEmail(),
+                supportingText = {
+                    if (!ValidateEmail()) {
+                        Text(
+                            modifier = Modifier.fillMaxWidth(),
+                            text = "Invalid email",
+                            color = MaterialTheme.colorScheme.error
+                        )
+                    }
+                },
+            )
+        }
+
+        Row(
+            modifier = Modifier
+                .padding(10.dp)
+        ) {
+            OutlinedTextField(
+                modifier = modifier
+                    .size(width = 400.dp, height = 50.dp)
+                    .fillMaxSize(),
+                value = password,
+                onValueChange = { newValue -> setPassword(newValue) },
+                placeholder = { Text("Password") },
+                visualTransformation = if (showPass) VisualTransformation.None else PasswordVisualTransformation(),
+                trailingIcon = {
+                    val (icon, iconColor) = if (showPass) {
+                        Pair(
+                            Icons.Filled.Visibility,
+                            Color(Color.Gray.value)
+                        )
+
+                    } else {
+                        Pair(Icons.Filled.VisibilityOff, Color(Color.Gray.value))
+                    }
+
+                    IconButton(onClick = { setShowPass(!showPass)  }) {
+                        Icon(
+                            icon,
+                            contentDescription = "Visibility",
+                            tint = iconColor
+                        )
+                    }
+                }
+            )
+        }
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
```

Comment:
By placing these modifiers here, you will override them if a modifier is supplied from the outside. Better chain them down in the `Button` itself.